ucm2/sof-soundwire/dmic.conf

```diff
@@ -0,0 +1,36 @@
+SectionDevice."Mic1" {
+	Comment "Digital Microphone"
+
+	Value {
+		# FIXME: value is not 100 to avoid duplicate priority reported
+		# by ucm-validator		
+		CapturePriority 101
+		CapturePCM "hw:${CardId},3"
```

Comment:
@plbossart I checked our tplgs of sdw with DMIC and found the PCM is not aligned.
1. on sof-cml-rt700 and sof-icl-rt700, it will use PCM2 for DMIC01 and PCM3 for DMIC16k.
2. on sof-tgl-rt5682 and sof-tgl-rt711-i2s-rt1308, it will use PCM3 for DMIC01 and PCM4 for DMIC16k. (The PIPELINE_PCM_ADD and PCM_PLAYBACK_ADD is not aligned in sof-tgl-rt5682)
3. on sof-tgl-sdw-max98373-rt5682, it will use PCM4 for DMIC01 and  PCM5 for DMIC16k (used for keyword detection)
So it seems the dmic PCM is not aligned. It is better to be aligned. What do you think? Or should we not touch the old platforms?

Also I checked rt715 codec DMIC, it will always use PCM4.

Reply:
I don't know how we can align @libinyang 
1. cml-rt700 does not have an amplifier
2. sof-tgl-rt5682 and sof-tgl-rt711-i2s-rt1308 have an external amplifier
3.sof-tgl-sdw-max98373-rt5682 has SoundWire.

The real problem is actually the device numbers, and in a number of cases we used 99 as a temporary solution (which has remained for 2 years). We also don't have consistency on the definition of pipelines, which means e.g. that the tgl-rt711 does not have the right controls for DMIC.

Reply:
> 1. an amplifier

@plbossart All sof-cml-rt700,  sof-icl-rt700, sof-tgl-rt5682, sof-tgl-rt711-i2s-rt1308, sof-tgl-sdw-max98373-rt5682 will use sof-sdw machine driver. So they will use dmic.conf based on this PR. But the DMIC PCM is not aligned. The will cause issues, like https://github.com/alsa-project/alsa-ucm-conf/pull/75#issuecomment-768471801. 

Mostly DMIC01 is used for 4 channels and DMIC16K is used for 2 channels. So the dmic.conf using CapturePCM "hw:${CardId},3" will cause 
1. platforms of sof-cml-rt700 and sof-icl-rt700 will use DMIC16k, which has 2 channels, 
2. platforms sof-tgl-rt5682 and sof-tgl-rt711-i2s-rt1308 will use DMIC01, which has 4 channels.
3. platforms of  sof-tgl-sdw-max98373-rt5682 will failed to find the PCM3, which will cause PA initialization fail.

As PCM num is separated from pipeline num, I suggest we use pcm #99 for DMIC01 (4 channel), pcm #100 for DMIC16k. What do you think?

Reply:
> @plbossart with this file
> in [dmesg](https://github.com/alsa-project/alsa-ucm-conf/files/5885757/dmesgj.txt):
> 
> ```
> [    6.392924] sof-audio-pci 0000:00:1f.3: tplg: load pcm DMIC16kHz 4
> [    6.392927] sof-audio-pci 0000:00:1f.3: can't find host comp to bind pcm
> [    6.392929] sof-audio-pci 0000:00:1f.3: error: can't bind pcm to host
> [    6.392931] sof-audio-pci 0000:00:1f.3: ASoC: DAI loading failed
> [    6.392931] sof-audio-pci 0000:00:1f.3: ASoC: topology: could not load header: -22
> ```
> 
> `ls /dev/snd`
> 
> ```
> seq  timer
> ```

Yes, I see the same error for the TGL reference platforms which looks very similar to your device.
https://sof-ci.01.org/sofpr/PR3790/build8029/devicetest/
We may have both a problem with device numbers and dailinks.